draft-2020-12/optional/format/date.js

```diff
@@ -0,0 +1,27 @@
+import spacetime from 'spacetime';
+
+/**
+ * Function to validate the date-time format using spacetime.
+ * @param {string} dateTimeString - The date-time string to validate.
+ * @returns {boolean} - True if the date-time string is valid, false otherwise.
+ */
+const validateDateTime = (dateTimeString) => {
+  if (typeof dateTimeString !== 'string') {
+    return true;
+  }
+  
+  const regex = /^\d{4}-(0[1-9]|1[0-2])-(0[1-9]|[12][0-9]|3[01])$/;
```

Comment:
I generally avoid declaring regular expressions within a function. Declaring it in the function could result in the regular expression needing to be compiled every time the function is called.

In practice, it's not likely to make a noticeable difference because the regex is fairly simple and it might end up getting cached and optimized by the JavaScript engine anyway, but I still think it's a good practice to move the regex out of the function.